### ISSUE TITLE
'$ firejail firefox -p' fails to launch

### ISSUE THREAD

**Issue by EauBureau** [netblue30/firejail#3837]
Since upgrading to Firefox 84 it no longer launches using `$ fireljail firefox -p`
Firefox's profile selector window is supposed to open.

Using `$ firejail firefox` does work. i.e Firefox opens and runs without profile selector option.

Using `$ firefox -p` does work. i.e. without firejail the profile selector window appears

Ubuntu 20.04 LTS
firejail version 0.9.62

```
$ firejail firefox -p
Reading profile /etc/firejail/firefox.profile
Reading profile /etc/firejail/whitelist-usr-share-common.inc
Reading profile /etc/firejail/firefox-common.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Warning: networking feature is disabled in Firejail configuration file
Parent pid 2676591, child pid 2676592
Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
Post-exec seccomp protector enabled
Seccomp list in: !chroot, check list: @default-keep, prelist: unknown,
Child process initialized in 73.01 ms
```

**Comment by rusty-snake** [comment]
I can not reproduce this.

**Comment by EauBureau** [comment]
Here's `jounralctl -f` for the following if they're of any use

```
$ firejail firefox -p

Dec 18 16:08:33 computer audit[154053]: SECCOMP auid=1000 uid=1000 gid=1000 ses=3 pid=154053 comm="GLXVsyncThread" exe="/usr/lib/firefox/firefox" sig=31 arch=c000003e syscall=312 compat=0 ip=0x7fa0cba3a89d code=0x0
Dec 18 16:08:33 computer kernel: audit: type=1326 audit(1608404913.711:70): auid=1000 uid=1000 gid=1000 ses=3 pid=154053 comm="GLXVsyncThread" exe="/usr/lib/firefox/firefox" sig=31 arch=c000003e syscall=312 compat=0 ip=0x7fa0cba3a89d code=0x0
```



```
$ firejail firefox

Dec 18 16:09:08 comptuer systemd[1]: Starting Clean php session files...
Dec 18 16:09:08 computer systemd[1]: phpsessionclean.service: Succeeded.
Dec 18 16:09:08 computer systemd[1]: Finished Clean php session files.

```




```
$ firefox -p

[nothing logged while the profile selector window appears and this while the actual Firefox window appears]

Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 2 threads of 2 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 2 threads of 2 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 2 threads of 2 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 2 threads of 2 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 2 threads of 2 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 2 threads of 2 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Successfully made thread 155586 of process 155368 owned by '1000' RT at priority 10.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 3 threads of 3 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 3 threads of 3 processes of 1 users.
Dec 18 16:10:46 computer rtkit-daemon[1288]: Supervising 3 threads of 3 processes of 1 users.
Dec 18 16:10:47 computer rtkit-daemon[1288]: Supervising 3 threads of 3 processes of 1 users.
Dec 18 16:10:47 computer rtkit-daemon[1288]: Supervising 3 threads of 3 processes of 1 users.

```

**Comment by rusty-snake** [comment]
> SECCOMP auid=1000 uid=1000 gid=1000 ses=3 pid=154053 comm="GLXVsyncThread" exe="/usr/lib/firefox/firefox" sig=31 arch=c000003e syscall=312 compat=0 ip=0x7fa0cba3a89d code=0x0

Maybe `firejail '--seccomp=!chroot,!kcmp' firefox -p` works.

**Comment by EauBureau** [comment]
> Maybe `firejail '--seccomp=!chroot,!kcmp' firefox -p` works.

Unfortunately it doesn't.

```
$ firejail '--seccomp=!chroot,!kcmp' firefox -p
Reading profile /etc/firejail/firefox.profile
Reading profile /etc/firejail/whitelist-usr-share-common.inc
Reading profile /etc/firejail/firefox-common.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Warning: networking feature is disabled in Firejail configuration file
Parent pid 313848, child pid 313849
Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
Post-exec seccomp protector enabled
Seccomp list in: !chroot, check list: @default-keep, prelist: unknown,
Child process initialized in 99.36 ms

```

```
$ journalctl -f

Dec 20 13:41:12 computer audit[313856]: SECCOMP auid=1000 uid=1000 gid=1000 ses=3 pid=313856 comm="GLXVsyncThread" exe="/usr/lib/firefox/firefox" sig=31 arch=c000003e syscall=312 compat=0 ip=0x7f6e2a34889d code=0x0
Dec 20 13:41:12 computer kernel: audit: type=1326 audit(1608562273.052:73): auid=1000 uid=1000 gid=1000 ses=3 pid=313856 comm="GLXVsyncThread" exe="/usr/lib/firefox/firefox" sig=31 arch=c000003e syscall=312 compat=0 ip=0x7f6e2a34889d code=0x0

```

**Comment by rusty-snake** [comment]
Gotcha! The `seccomp` in the profile overrides the `seccomp` on the command line. See "Seccomp list in: !chroot, check list: @default-keep, prelist: unknown,". `firejail '--seccomp=!chroot,!kcmp' '--ignore=seccomp' firefox -p`.

**Comment by EauBureau** [comment]
Great job. Thank you.

What does this mean: for now I use the new command above, but down the road when a newer version of firejail is released the previous simple `firejail firefox -p` will work again?

Or is this issue somehow just relevant to my system's setup?

**Comment by rusty-snake** [comment]
> newer version of firejail is released
> Or is this issue somehow just relevant to my system's setup?

#3219

> for now I use the new command above

You can also add the `!kcmp` to your firefox.profile/firefox.local, so `firejail firefox -p` works.

> Does running with the workaround command diminish the security at all?

It allows the `kcmp` syscall.

**Comment by EauBureau** [comment]
> You can also add the `!kcmp` to your firefox.profile/firefox.local, so `firejail firefox -p` works.
That's perfect, thanks.

Will I close this or do you want to keep it open?

**Comment by rusty-snake** [comment]
Yes, I think we can closes here, there is still #3219 for a general solution on kcmp in mesa+amd.
